modules/uhttp/filters.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package uhttp
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"golang.org/x/net/context"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// Filter applies filters on requests, request contexts or responses such as
+// adding tracing to the context
+type Filter interface {
+	Apply(w http.ResponseWriter, r *http.Request, next http.Handler)
+}
+
+// FilterFunc is an adaptor to call normal functions to apply filters
+type FilterFunc func(w http.ResponseWriter, r *http.Request, next http.Handler)
+
+// Apply implements Apply from the Filter interface and simply delegates to the function
+func (f FilterFunc) Apply(w http.ResponseWriter, r *http.Request, next http.Handler) {
+	f(w, r, next)
+}
+
+type tracerFilter struct {
+	tracer opentracing.Tracer
+}
+
+func (t tracerFilter) Apply(w http.ResponseWriter, r *http.Request, next http.Handler) {
+	operationName := r.Method
+	carrier := opentracing.HTTPHeadersCarrier(r.Header)
+	spanCtx, err := t.tracer.Extract(opentracing.HTTPHeaders, carrier)
+	if err != nil && err != opentracing.ErrSpanContextNotFound {
+		log.Printf("Malformed inbound tracing context: %s", err.Error())
```

Comment:
This shouldn't be using stdlib log. Use `ulog.Logger()` or preferrably a module-scoped logger